app/controllers/sessions_controller.rb

```diff
@@ -2,54 +2,83 @@ class SessionsController < ApplicationController
   def new
   end
 
+  def shop_owner_login
+  end
+
   def create
     if request.env["omniauth.auth"]
       oauth_user = OauthUser.new(request.env["omniauth.auth"])
       user = oauth_user.login_or_create
       session[:user_id] = user.id
 
       return redirect_to_user_intended unless session[:user_intended].nil?
-      redirect_to root_path, notice: "Welcome back, #{current_user.first_name}"
+      redirect_to root_path, notice: welcome_user
     else
-      process_form_login
+      process_form_login("RegularUser")
     end
   end
 
+  def shop_owner_create
+    process_form_login("ShopOwner")
+  end
+
   def destroy
     @current_user = nil
     session.delete :user_id
-    redirect_to root_path, notice: "Logged out Successfully!"
+    redirect_to login_path, notice: logout
+  end
+
+  def shop_owner_destroy
```

Comment:
**Critical - Code Quality**
The logic implemented in this method is similar the one implemented in `destroy` on line 25.

Please refactor so that we do not have duplicate code.